### DISCUSSION TITLE
전체 코치 조회 관련 API 논의

### DISCUSSION THREAD

**Discussion by her0807** [woowacourse-teams/2022-ternoko#96] [Q&A] [answered]
### 문제점
현재 코치 findAll 인데 API 가  `api/reservations/coaches ` 로 되어 있습니다.

###  제안 사항
저는 reservations 을 붙이고 싶지 않은데요,  이유는 크게 두가지 입니다. 

1) 코치 목록 조회라는 행위만 봤을때 reservations 가 붙어야할까요 ? `예약/코치들 ` 보다 `코치들` 이 더 명확해 보입니다.
2) 현재 CoachController 와 CrewController 로 분기될 예정인데, 앞에 reservations 가 붙어서  CoachController 에 있는 @RequestMapping("/api/coachs") 때문에  아래로 들어갈 수 없습니다.

의견 부탁드립니다. ✨

```java
@RequestMapping("/api")
public class MemberController {

	private final CoachService coachService;

	@GetMapping("/reservations/coaches")
	public ResponseEntity<CoachesResponse> findCoaches() {
		return ResponseEntity.ok(coachService.findCoaches());
	}
```

**Comment by her0807** [comment]
@lokba  @HyeonbinSa @sanaandmomo  @Juhyung990122  @soominsohn  @dongho108

**Comment by sanaandmomo** [comment]
저도 수달의 의견에 동의합니다.
왜냐하면 현재 url은 예약건이 잡혀있는 코치들을 조회한다라는 의미로 더 해석이 된다고 생각하기 때문입니다.

**Comment by Juhyung990122** [comment]
저도 동의합니다! 빼는게 더 명확해보이네요🤔 컨트롤러 분리과정에서 같이 정리하면 좋을 것 같습니다

**Comment by HyeonbinSa** [comment, marked as answer]
우선 왜 저렇게 됐는지를 말씀드리면 코치 선택 후 시간 선택하는 페이지로 넘어가도록 기획했었기때문에 '예약 과정 중 코치 목록 조회' 의미로 저렇게 사용했었습니다.
그래서 어제 현재 방식과 맞지 않으니 coach id를 바디에 넣고 목록 조회는 '/api/coaches'로 사용하기로 했습니다.
[터놓고 API-코치목록 조회](https://www.notion.so/3-API-3d4106383e3c41c5a712569befc3705f#5e4d7e039ba74ced9924859111cc3935)

좋은 의견이고 앞으로 관련해서 coach, crew, reservation 컨트롤러에 어떻게 나눌지 고민해보면 좋을거같습니다.

**Comment by dongho108** [comment]
저도 동의합니당

이거 데모데이2 발표때 api 정리했을때 이렇게 정리했고, 어제 api 명세 정할때도 `/api/coches`로 하기로 했어요!!
api 명세 바뀐거 한번에 오늘 수정하겠습니당

**Comment by lokba** [comment]
1번의 이유에서 수정한 방식이 더 좋아보이네요! 동의합니다.

**Comment by soominsohn** [comment]
수달 좋은 의견 고마워요! 앞으로도 api uri 설계할 때 많은 고민이 필요하겠군요! 💯
